counterparty-core/counterpartycore/lib/transaction.py

```diff
@@ -959,3 +954,815 @@ def get_dust_return_pubkey(source, provided_pubkeys, encoding):
+    unspent_tx_hash=None,
+    custom_inputs=None,
+    dust_return_pubkey=None,
+    disable_utxo_locks=False,
+    extended_tx_info=False,
+    p2sh_source_multisig_pubkeys=None,
+    p2sh_source_multisig_pubkeys_required=None,
+    p2sh_pretx_txid=None,
+    old_style_api=True,
+    segwit=False,
+    api_v1=False,
+):
+    """Create and return a transaction."""
+
+    # Get provided pubkeys.
+    if isinstance(pubkey, str):
+        provided_pubkeys = [pubkey]
+    elif isinstance(pubkey, list):
+        provided_pubkeys = pubkey
+    elif pubkey is None:
+        provided_pubkeys = []
+    else:
+        raise exceptions.TransactionError("Invalid pubkey.")
+
+    # Get additional pubkeys from `source` and `destination` params.
+    # Convert `source` and `destination` to pubkeyhash form.
+    for address_name in ["source", "destination"]:
+        if address_name in params:
+            address = params[address_name]
+            if isinstance(address, list):
+                # pkhshs = []
+                # for addr in address:
+                #    provided_pubkeys += script.extract_pubkeys(addr)
+                #    pkhshs.append(script.make_pubkeyhash(addr))
+                # params[address_name] = pkhshs
+                pass
+            else:
+                provided_pubkeys += script.extract_pubkeys(address)
+                params[address_name] = script.make_pubkeyhash(address)
+
+    # Check validity of collected pubkeys.
+    for pubkey in provided_pubkeys:
+        if not script.is_fully_valid(binascii.unhexlify(pubkey)):
+            raise script.AddressError(f"invalid public key: {pubkey}")
+
+    compose_method = sys.modules[f"counterpartycore.lib.messages.{name}"].compose
+    compose_params = inspect.getfullargspec(compose_method)[0]
+    missing_params = [p for p in compose_params if p not in params and p != "db"]
+    if api_v1:
+        for param in missing_params:
+            params[param] = None
+    else:
+        if len(missing_params) > 0:
+            default_values = get_default_args(compose_method)
+            for param in missing_params:
+                if param in default_values:
+                    params[param] = default_values[param]
+                else:
+                    raise exceptions.ComposeError(
+                        f"missing parameters: {', '.join(missing_params)}"
+                    )
+
+    # dont override fee_per_kb if specified
+    if fee_per_kb is not None:
+        estimate_fee_per_kb = False
+    else:
+        fee_per_kb = config.DEFAULT_FEE_PER_KB
+
+    if "extended_tx_info" in params:
+        extended_tx_info = params["extended_tx_info"]
+        del params["extended_tx_info"]
+
+    if "old_style_api" in params:
+        old_style_api = params["old_style_api"]
+        del params["old_style_api"]
+
+    if "segwit" in params:
+        segwit = params["segwit"]
+        del params["segwit"]
+
+    tx_info = compose_method(db, **params)
+    initialise(db)
```

Comment:
For my own edification, why is this the right place to initialize the UTXO_LOCKS?

Reply:
Oh, a more points here.  `initialise` doesnt take a db.  also, it is currently called in `server.start_all`.  perhaps it was a mistake for me to put it there but I believe this is a no-op.

Reply:
hum.. yeah good catch! I don't know what this line is doing here! perhaps a merge error! will remove it.